racksdb/web/ui/src/components/InfrastructureTable.vue

```diff
@@ -10,364 +10,239 @@ import type {
   Infrastructure,
   NodeEquipment,
   NetworkEquipment,
-  StorageEquipment
+  StorageEquipment,
+  MiscEquipment
 } from '@/composables/RacksDBAPI'
-import { ref } from 'vue'
-
-const rack = ref()
-const showRack = ref(false)
-const showInfrastructureRacks = ref(true)
+import { ref, onMounted } from 'vue'
+import {
+  BarsArrowDownIcon,
+  BarsArrowUpIcon,
+  ChevronDownIcon,
+  ChevronUpIcon
+} from '@heroicons/vue/24/outline'
+import InfrastructurePopUp from '@/components/InfrastructurePopUp.vue'
+
+const rackNames: Ref<Array<string>> = ref([])
```

Comment:
You use an array `rackNames` whose indexes are used to determine the display boolean of the corresponding rack in `rackDisplayStates` array. You should better use a `Record<string, boolean>` object to associate racks names to their boolean display state.

Reply:
I made the changes and remove rackNames, rackDisplayStates and use Reccord.